lib/streamlit/elements/arrow.py

```diff
@@ -568,17 +568,15 @@ def dataframe(
             proto.id = compute_and_register_element_id(
                 "dataframe",
                 user_key=key,
+                form_id=proto.form_id,
                 data=proto.data,
                 width=width,
                 height=height,
                 use_container_width=use_container_width,
                 column_order=proto.column_order,
                 column_config=proto.columns,
-                key=key,
                 selection_mode=selection_mode,
                 is_selection_activated=is_selection_activated,
-                form_id=proto.form_id,
-                page=ctx.page_script_hash if ctx else None,
```

Comment:
was there a reason why here `ctx.page_script_hash` was used and not `ctx.active_script_hash`? because now the latter is used. But probably was just an oversight, right?

Reply:
The usage of `page_script_hash` before was wrong